hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/HapiSpecOperation.java

```diff
@@ -354,7 +356,9 @@ protected Transaction finalizedTxn(
                 .andThen(opDef);
 
         setKeyControlOverrides(spec);
-        final List<Key> keys = signersToUseFor(spec);
+        List<Key> keys = signersToUseFor(spec);
+        Iterables.removeIf(keys, Predicates.isNull());
```

Comment:
Why would there be `null` keys here?

Reply:
While testing I had the problem of having `null` keys here. I really can't remember the exact case but this was the reason I added this line.

Dealing with the nulls inside `signersToUseFor` makes more sense, I agree. Fixed 👍